Shared/sdk/SharedUtil.Hash.hpp

```diff
@@ -818,5 +818,22 @@ namespace SharedUtil
 
         out->assign((char*)buffer, numPasses * 4);
         delete[] buffer;
+
+        if (trimPaddingAtEnd)
+        {
+            // Check last block for 0 padding, and delete
+
+            auto iter = out->end() - 1;
+            if (*iter != 0)
+                return;
+
+            const auto blockEnd = out->end() - 4;
+            while (iter > blockEnd && *--iter == 0);
```

Comment:
Please use more lines

Reply:
I feel like this is perfectly readable. Not that anyone reads that code anyways.

Reply:
This requires less thinking
```
while (iter > blockEnd || *iter == 0)
    iter--;
```